packages/pigeon/example/app/linux/messages.g.h

```diff
@@ -0,0 +1,99 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+// Autogenerated from Pigeon (v11.0.1), do not edit directly.
+// See also: https://pub.dev/packages/pigeon
+
+#ifndef PIGEON_MESSAGES_G_H_
+#define PIGEON_MESSAGES_G_H_
+
+#include <flutter_linux/flutter_linux.h>
+
+G_BEGIN_DECLS
+
+typedef enum { MY_CODE_ONE = 0, MY_CODE_TWO = 1 } MyCode;
+
+G_DECLARE_FINAL_TYPE(MyMessageData, my_message_data, MY, MESSAGE_DATA, GObject)
+
+MyMessageData* my_message_data_new(MyCode code, FlValue* data);
+
+MyMessageData* my_message_data_new_full(const gchar* name,
+                                        const gchar* description, MyCode code,
+                                        FlValue* data);
+
+const gchar* my_message_data_get_name(MyMessageData* object);
+
+const gchar* my_message_data_get_description(MyMessageData* object);
+
+MyCode my_message_data_get_code(MyMessageData* object);
+
+FlValue* my_message_data_get_data(MyMessageData* object);
+
+G_DECLARE_FINAL_TYPE(MyExampleHostApi, my_example_host_api, MY,
+                     EXAMPLE_HOST_API, GObject)
+
+typedef struct {
+  void (*get_host_language)(
+      MyExampleHostApi* object,
+      FlBasicMessageChannelResponseHandle* response_handle, gpointer user_data);
+  void (*add)(MyExampleHostApi* object,
+              FlBasicMessageChannelResponseHandle* response_handle, int64_t a,
```

Comment:
Currently sync and async methods are using the same design. Is there a standard GObject design pattern for synchronous methods that can fail and return an error? We could use this approach if we have to, but being able to use standard `return` semantics for things that don't need to be async is one of the nice things about Pigeon, because it's a less error-prone system (i.e., you can't accidentally not call a response, or call it twice).

Reply:
I found it somewhat difficult to make the sync calls with the C code, so for this first iteration just made all returns the same. We would have to make a new class like ErrorOr in the Windows support. I'll have another go at this, hopefully we can find some API that forces a sync response without being too heavy.

Reply:
Made objects for returning sync values.